counsel.el

```diff
@@ -1438,22 +1438,42 @@ This function should set `ivy--old-re'."
 
 (defun counsel-git-grep-action (x)
   "Go to occurrence X in current Git repository."
-  (when (string-match "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'" x)
+  (when (string-match (counsel-git-grep-match-regex) x)
     (let ((file-name (match-string-no-properties 1 x))
           (line-number (match-string-no-properties 2 x)))
```

Comment:
Have a lot and see how like this. For for `rx-define`, I had a hard time recreating the original regex. Specifically, mine looks as follows

```
(rx-define grouped-regex-matches
  (seq string-start (group (zero-or-more any))
       ":" (group (+ digit)) ":"
       (group (* any)) string-end))
(rx grouped-regex-matches); "\\`\\(.*\\):\\([[:digit:]]+\\):\\(.*\\)\\'"
;; original  ----------------------> "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'"
```

Notice the additional question mark in the original regex. I couldn't get rx to generate it like this.

Reply:
Sorry, I wasn't very clear and misled you:

> [ As an aside, my preferred way of defining a regexp for reuse since Emacs 27 would be via `rx-define`. ]

By this I meant that, in my ideal world, we would use `rx-define` to name relevant regexps, and also better guarantee their correctness (writing regexps, especially in the presence of Elisp's unique regexp constructs, as strings is all too error-prone in practice).

There are two reasons we can't use `rx-define` (today):
- it did not exist before Emacs 27
- IIRC @abo-abo is not a fan of `rx`

> For for `rx-define`, I had a hard time recreating the original regex.

Have you tried the [`xr`](https://elpa.gnu.org/packages/xr.html) package?  It can parse regexp strings into equivalent `rx` constructs.  For example:

```el
(xr "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'" 'terse)
;; => (: bos (group (*? nonl))
;;       ":" (group (+ (in "0-9")))
;;       ":" (group (* nonl))
;;       eos)
```

> Notice the additional question mark in the original regex. I couldn't get rx to generate it like this.

That's because you're using `zero-or-more` without wrapping it in `minimal-match`.  Personally I prefer the terser `*?` as seen in the `xr` translation above.

Reply:
Ok, rx is out! :)

I agree, rx is easier to read than a bunch of globs. TBH though, I've always found regex in the emacs world a little jarring.

I had seen xr, but managed to forget about it between commits. 

Let me know what other changes you'd like and I'll keep my eyes out for the next step in the Copyright Assignment process :crossed_fingers: